File: app/src/main/java/com/cookplan/add_ingredient_view/ProductCategoriesSpinnerAdapter.java
package com.cookplan.add_ingredient_view;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.SpinnerAdapter;
import android.widget.TextView;

import com.cookplan.R;
import com.cookplan.RApplication;
import com.cookplan.models.ProductCategory;

import java.util.List;

/**
 * Created by DariaEfimova on 20.03.17.
 */

public class ProductCategoriesSpinnerAdapter extends ArrayAdapter<ProductCategory> implements SpinnerAdapter {
    private List<ProductCategory> itemsAll;

    public ProductCategoriesSpinnerAdapter(Context context, List<ProductCategory> itemsAll) {
        super(context, R.layout.product_category_spinner_layout, itemsAll);
        this.itemsAll = itemsAll;
    }

    @NonNull
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        View v = convertView;

        if (v == null) {
            LayoutInflater vi = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            v = vi.inflate(R.layout.product_category_dropdown_layout, null);
        }

        ProductCategory category = itemsAll.get(position);
        if (category != null) {
            TextView categoryNameTextView = (TextView) v.findViewById(R.id.category_product_item_name);
            if (categoryNameTextView != null) {
                categoryNameTextView.setText(category.toString());
                categoryNameTextView.setTextColor(ContextCompat.getColor(RApplication.Companion.getAppContext(),
                                                                         category.getColorId()));
            }
        }
        return v;
    }

    @Override
    public View getDropDownView(int position, View convertView, ViewGroup parent) {
        View v = convertView;

        ProductCategory category = itemsAll.get(position);

        if (category != null) {
            if (v == null) {
                LayoutInflater vi = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
                v = vi.inflate(R.layout.product_category_spinner_layout, null);
            }
            TextView categoryNameTextView = (TextView) v.findViewById(R.id.category_product_item_name);
            if (categoryNameTextView != null) {
                categoryNameTextView.setText(category.toString());
            }
            View categoryView = v.findViewById(R.id.main_view);
            if (categoryView != null) {
                categoryView.setBackgroundResource(category.getColorId());
            }
        }

        return v;
    }
}

File: app/src/main/java/com/cookplan/add_ingredient_view/ProductCategoriesSpinnerAdapter.kt
package com.cookplan.add_ingredient_view

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.SpinnerAdapter
import android.widget.TextView
import androidx.core.content.ContextCompat
import com.cookplan.R
import com.cookplan.models.ProductCategory

/**
 * Created by DariaEfimova on 20.03.17.
 */

class ProductCategoriesSpinnerAdapter(context: Context, private val itemsAll: List<ProductCategory>) : ArrayAdapter<ProductCategory>(context, R.layout.product_category_spinner_layout, itemsAll), SpinnerAdapter {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val vi = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        val view = convertView ?: vi.inflate(R.layout.product_category_dropdown_layout, null)


        val category = itemsAll[position]
        val categoryNameTextView = view.findViewById<TextView>(R.id.category_product_item_name)
        categoryNameTextView?.setText(category.getNameResourceId())
        categoryNameTextView?.setTextColor(ContextCompat.getColor(parent.context, category.colorId))
        return view
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {

        val vi = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = convertView ?: vi.inflate(R.layout.product_category_spinner_layout, null)

        val category = itemsAll[position]

        val categoryNameTextView = view.findViewById<TextView>(R.id.category_product_item_name)
        categoryNameTextView?.setText(category.getNameResourceId())
        val categoryView = view.findViewById<View>(R.id.main_view)
        categoryView?.setBackgroundResource(category.colorId)

        return view
    }
}